dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsHelper.java

```diff
@@ -95,16 +96,32 @@
             appDescriptors = appDescriptors.stream().filter(appDescriptor -> appDescriptor.getName().matches(regexFilter)).collect(
                     Collectors.toList());
         }
-        final Set<String> hostIdentifiers = appsAPI.appKeysByHost().keySet();
+        final Set<String> siteIdentifiers = appsAPI.appKeysByHost().keySet();
         for (final AppDescriptor appDescriptor : appDescriptors) {
             final String appKey = appDescriptor.getKey();
-            final long configurationsCount = appsAPI
-                    .filterSitesForAppKey(appKey, hostIdentifiers, user).size();
-            views.add(new AppView(appDescriptor, configurationsCount));
+            final int configurationsCount = appsAPI.filterSitesForAppKey(appKey, siteIdentifiers, user).size();
+            final int sitesWithWarning = computeWarningsBySite(appDescriptor, siteIdentifiers, user);
+            views.add(new AppView(appDescriptor, configurationsCount, sitesWithWarning));
         }
         return views.stream().sorted(compareByCountAndName).collect(CollectionsUtils.toImmutableList());
     }
 
+    /**
+     * Computes the number of warnings regardless of site under the given app-descriptor
+     * @param appDescriptor
+     * @param sitesWithConfigurations
+     * @param user
+     * @return sum or warnings for the given app-descriptor. Regardless of site.
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    private int computeWarningsBySite(final AppDescriptor appDescriptor,
+            final Set<String> sitesWithConfigurations, final User user)
+            throws DotDataException, DotSecurityException {
+        final Map<String, Map<String, List<String>>> warningsBySite = appsAPI.computeWarningsBySite(appDescriptor, sitesWithConfigurations, user);
```

Comment:
I think we can create a class to handle this 'Map<String, Map<String, List<String>>>', it can do the code more readable, this class could have methods like

- Map<String, List<String>> getXXX(String)
- or Even List<String> getXXX(String, String)